docker-app/qfieldcloud/core/filters.py

```diff
@@ -0,0 +1,44 @@
+import django_filters
+from django.db import models
+
+from qfieldcloud.core.models import Project, ProjectQueryset
+
+
+class IncludePublicChoices(models.IntegerChoices):
+    EXCLUDE = 0, "Exclude"
+    INCLUDE = 1, "Include"
+
+
+class ProjectFilterSet(django_filters.FilterSet):
+    owner = django_filters.CharFilter(field_name="owner__username", lookup_expr="exact")
+    include_public = django_filters.ChoiceFilter(
+        label="Include public projects",
+        choices=IncludePublicChoices.choices,
+        method="filter_include_public",
+    )
+
+    def filter_include_public(self, queryset, name, value):
+        if value != "1":
+            queryset = queryset.exclude(
+                user_role_origin=ProjectQueryset.RoleOrigins.PUBLIC
+            )
+
+        return queryset
+
+    def filter_queryset(self, queryset):
+        include_public = "0"
+        if (
+            "include-public" in self.form.data
+            and self.form.data["include-public"] != ""
+        ):
+            include_public = self.form.data["include-public"]
+
+        if "include_public" in self.form.cleaned_data:
+            if self.form.cleaned_data["include_public"] == "":
+                self.form.cleaned_data["include_public"] = include_public
+
+        return super().filter_queryset(queryset)
```

Comment:
```suggestion
    def filter_queryset(self, queryset: models.QuerySet[Project]) -> models.QuerySet[Project]:
        if self.form.cleaned_data.get("include_public") != "":
            return super().filter_queryset(queryset)

        # This is a workaround to make the filter work with the `include-public` parameter,
        # since Django does not have a good support for hyphens in field names and it was previously used in the API.
        self.form.cleaned_data["include_public"] = self.form.data["include-public"]

        return super().filter_queryset(queryset)
```

I think this is simpler to read and maintain and doing the same thing. Tried it locally, seems to work fine.

Reply:
Good.